tokio/src/io/driver/mod.rs

```diff
@@ -237,11 +198,14 @@ impl Handle {
     /// # Panics
     ///
     /// This function panics if there is no current reactor set.
+    #[cfg(all(feature = "io-driver", not(loom)))]
     pub(super) fn current() -> Self {
-        CURRENT_REACTOR.with(|current| match *current.borrow() {
-            Some(ref handle) => handle.clone(),
-            None => panic!("no current reactor"),
-        })
+        ThreadContext::io_handle().expect("no current reactor")
+    }
+
+    #[cfg(any(not(feature = "io-driver"), loom))]
+    pub(super) fn current() -> Self {
```

Comment:
I don't think we need this branch?

Reply:
I would love to get rid of this but I couldn't find a way around `runtime::io::variant`. 
With `#[cfg(all(feature = "io-driver", not(loom)))]`, the `Handle` type is
`pub(crate) type Handle = Option<driver::Handle>;`
With `#[cfg(any(not(feature = "io-driver"), loom))]`
`pub(crate) type Handle = ();`
This fails to compile with loom because `expect` needs an `Option`. 

Definitely open to suggestions here :)

Reply:
You're right! bleh that is confusing, we can prob clean that up in a follow up PR but not sure what @carllerche wants to do with that.